miner/tasks.py

```diff
@@ -0,0 +1,93 @@
+import os
+import time
+import hashlib
+import math
+from functools import reduce
+from celery import signals
+from transactions.models import Transaction
+from transactions.models import TransactionOutput
+from blocks.models import Block
+
+
+def gen_reward_transaction(val, blk_h):
+    # address for the block reward
+    reward_addr = os.environ.get("REWARD_ADDR", "a5a5da92a90a9ee3b1c640b18be05be07d335127a231ce631b503bfca76876dd3eb871dfaabd5aafe1f749a828577a9d")
+    # hash of the reward output
+    reward_output_hash = hashlib.sha1((str(val) + reward_addr + str(blk_h)).encode()).hexdigest()
+    # hash of the transaction(i.e the block reward transaction)
+    tx_hash = hashlib.sha1(('01'+reward_output_hash).encode()).hexdigest()
+    # save the block reward transaction
+    tx = Transaction.objects.create(hash=tx_hash, tx_inputs_count=0, tx_outputs_count=1)
+    # save the block reward transaction output
+    tout = TransactionOutput.objects.create(blk_height=blk_h, value=val, own_addr=reward_addr, hash=reward_output_hash, gen_transaction=tx, gen_transaction_index=0)
+    # return the block reward transaction
+    return tx
+
+
+@signals.worker_ready.connect
+def mine(**kwargs):
```

Comment:
Add docstrings on what the signal is doing. Very important in this case.

Reply:
done

Reply:
Add signals to a file called signals.py 
The signals go there.